src/domain-services/flows/flow-service.ts

```diff
@@ -0,0 +1,335 @@
+    args: IGetUniqueFlowsArgs
+  ): Promise<UniqueFlowEntity[]> {
+    const { databaseConnection, orderBy, conditions, whereClauses } = args;
+
+    let query = databaseConnection!
+      .queryBuilder()
+      .distinct('id', 'versionID', orderBy.column) // Include orderBy.column in the distinct selection
+      .from('flow')
+      .whereNull('deletedAt');
+    if (orderBy.raw) {
+      query = query.orderByRaw(orderBy.raw);
+    } else {
+      query = query.orderBy(orderBy.column, orderBy.order);
+    }
+
+    if (conditions) {
+      query = applySearchFilters(query, conditions);
+    }
+    if (whereClauses) {
+      query = query.andWhere(prepareCondition(whereClauses));
+    }
+
+    const flows = await query;
+    const mapFlowsToUniqueFlowEntities = flows.map((flow) => ({
+      id: flow.id,
+      versionID: flow.versionID,
+    }));
+
+    return mapFlowsToUniqueFlowEntities;
+  }
+
+  async getFlowsRaw(
+    databaseConnection: Knex,
+    whereClauses: FlowWhere,
+    orderBy: FlowOrderByCond | null
+  ): Promise<FlowInstance[]> {
+    let query = databaseConnection!
+      .queryBuilder()
+      .select('*')
+      .from('flow')
+      .whereNull('deletedAt');
+
+    if (orderBy?.raw) {
+      query = query.orderByRaw(orderBy.raw);
+    } else if (orderBy?.order) {
+      query = query.orderBy(orderBy.column, orderBy.order);
+    }
+
+    if (whereClauses) {
+      query = query.andWhere(prepareCondition(whereClauses));
+    }
+
+    const flows: FlowInstance[] = await query;
+
+    return flows;
+  }
+
+  async getFlowsCount(databaseConnection: Knex, conditions: FlowWhere) {
+    // Since this is the only place where we need to use the count method
+    // Instead of implementing on the models, we can use the queryBuilder directly
+    // Why use the count method? Because it's faster than fetching all the data
+    // and since we are filtering and paginating it makes no sense to fetch all the data
+    // just to do the count - this approach is faster and more efficient
+    const query = databaseConnection
+      .queryBuilder()
+      .count('*')
+      .from('flow')
+      .where(prepareCondition(conditions));
+
+    const [{ count }] = await query;
+
+    return Number(count);
+  }
+
+  async getFlowIDsFromEntity(
+    dbConnection: Knex,
+    orderBy: FlowOrderBy
+  ): Promise<UniqueFlowEntity[]> {
+    const entity = orderBy.subEntity ?? orderBy.entity;
+
+    // Get the entity list
+    const mappedOrderBy = mapOrderByToEntityOrderBy(orderBy);
+    const orderByRaw = `"${mappedOrderBy.column}" ${mappedOrderBy.order} nulls last`;
+
+    // 'externalReference' is a special case
+    // because it does have a direct relation with flow
+    // and no direction
+    if (entity === 'externalReference') {
+      const entityList = await dbConnection
+        .queryBuilder()
+        .distinct('flowID', 'versionID', mappedOrderBy.column)
+        .from('externalReference')
+        .orderByRaw(orderByRaw);
```

Comment:
Across this entire PR, you are violating a lot of the practices we've established around how DB models are used and how we query data. First, you shouldn't have exposed query builder at all. You should remove it and rewrite all the code to use `models` instead. Models already expose query builder, although we are very reluctant to use it and should only use it very rarely, when our models library cannot support such query. This would be an example how to use it:
```ts
await models.planTag.find({
  where: {
    [models.Cond.BUILDER]: (builder) => {
      builder.distinct(...);
    },
  },
});
```
However, the biggest reason we have our own model library is to have strong typing of our models. With using query builder from Knex directly, you lose that. And we don't want to give up on strong and strict typing.

What you should do instead is extend the model library to do what you want, or find a way around, for example, you can implement your own `DISTINCT` functionality by processing the "regular" results.

Then, you violate established practices even more further down in this method by using `JOIN`s. We have moved away from using joins a long time ago for 2 reasons:
1. Some of the joins become so complex that they take long time to execute, clogging the database and causing systemic slowdowns or plain crashes
2. We cannot have strong typing in Typescript when using joins. This is the biggest problem with all ORMs that are available in Node.js ecosystem. This is also one of the reasons we stopped using Sequelize (ORM) and moved to Knex (query builder).

I am using this code block just as an example of one place where you've violated established practices, but you should fix the problem elsewhere too.